lte/gateway/python/magma/mobilityd/tests/test_ip_allocator_dhcp.py

```diff
@@ -0,0 +1,316 @@
+"""
+Copyright 2022 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from datetime import datetime, timedelta
+from ipaddress import IPv4Address, IPv4Network
+from typing import Any, List
+from unittest.mock import MagicMock, patch
+
+import fakeredis
+import freezegun
+import pytest
+from magma.mobilityd.dhcp_desc import DHCPDescriptor, DHCPState
+from magma.mobilityd.ip_allocator_dhcp import (
+    DHCP_HELPER_CLI_PATH,
+    IPAllocatorDHCP,
+)
+from magma.mobilityd.ip_descriptor import IPDesc, IPState, IPType
+from magma.mobilityd.mac import MacAddress, sid_to_mac
+from magma.mobilityd.mobility_store import (
+    AssignedIpBlocksSet,
+    MobilityStore,
+    defaultdict_key,
+    ip_states,
+)
+
+SID = "IMSI123456789"
+MAC = MacAddress(sid_to_mac(SID).lower())
+MAC2 = MacAddress("01:23:45:67:89:ab")
+IP = "1.2.3.4"
+SERVER_IP = "5.6.7.8"
+SUBNET = "24"
+IP_NETWORK = "1.2.3.0/" + SUBNET
+IP_NETWORK_2 = "1.2.4.0/" + SUBNET
+VLAN = "0"
+LEASE_EXPIRATION_TIME = 4
+FROZEN_TEST_TIME = "2021-01-01"
+
+
+@pytest.fixture
+def dhcp_desc_fixture() -> DHCPDescriptor:
+    with freezegun.freeze_time(FROZEN_TEST_TIME):
+        return DHCPDescriptor(
+            mac=MAC,
+            ip=IP_NETWORK,
+            vlan=VLAN,
+            state=DHCPState.ACK,
+            state_requested=DHCPState.REQUEST,
+            lease_expiration_time=LEASE_EXPIRATION_TIME,
+        )
+
+
+@pytest.fixture
+def ip_allocator_fixture() -> IPAllocatorDHCP:
+    client = fakeredis.FakeStrictRedis()
+
+    ip_allocator = IPAllocatorDHCP(
+        store=MobilityStore(client),
+        lease_renew_wait_min=LEASE_EXPIRATION_TIME,
+        start=False,
+    )
+
+    yield ip_allocator
+
+    ip_allocator._monitor_thread_event.set()
+    ip_allocator._monitor_thread.join()
+
+
+@pytest.fixture
+def ip_allocator_dhcp_fixture(
+        ip_allocator_fixture: IPAllocatorDHCP,
+        dhcp_desc_fixture: DHCPDescriptor,
+) -> IPAllocatorDHCP:
+    ip_allocator_fixture._store.dhcp_store[
+        dhcp_desc_fixture.mac.as_redis_key(dhcp_desc_fixture.vlan)
+    ] = dhcp_desc_fixture
+
+    yield ip_allocator_fixture
+
+
+@pytest.fixture
+def ip_desc_fixture() -> IPDesc:
+    return IPDesc(
+        ip=IPv4Address(IP),
+        state=IPState.ALLOCATED,
+        vlan_id=int(VLAN),
+        ip_block=IPv4Network(IP_NETWORK),
+        ip_type=IPType.DHCP,
+        sid=SID,
+    )
+
+
+# def create_subprocess_mock() -> MagicMock:
+#     m = MagicMock()
+#     m.returncode = 0
+#     m.stdout = """{"lease_expiration_time": %s}""" % LEASE_EXPIRATION_TIME
+#     return m
+
```

Comment:
Is this block relevant or can it be deleted?

Reply:
It was deleted, thanks for spotting it!